src/almanack/entropy.py

```diff
@@ -0,0 +1,39 @@
+"""
+This module finds the amount of entropy on a file based level
+"""
+
+import math
+
+from .git_parser import calculate_loc_changes
+
+
+def calculate_shannon_entropy(
+    repo_path: str, source_commit: str, target_commit: str, file_names: list[str]
+) -> dict[str, float]:
+    """
+    Calculates the Shannon entropy for changes in specified files between two commits.
```

Comment:
Double checking: is this strictly Shannon entropy or was some other resource used to help build the work? My worry is that Shannon may not have specifically mentioned lines of code changes and that we might need to adjust the reference (that it is Shannon inspired).